File: Sorts & Search & Shuffling Algorithms/Elementry Search/Ternary Search/IterativeTernary.java
import java.util.ArrayList;

public class IterativeTernary {
    public static <T extends Comparable<T>> int Search(ArrayList<T> iterable, int first_index, int last_index, T key){
        while (last_index >= first_index) {
            int mid1 = first_index + (last_index - first_index) / 3;
            int mid2 = last_index - (last_index - first_index) / 3;
            if (iterable.get(mid1).equals(key)) {
                return mid1;
            }
            if (iterable.get(mid2).equals(key)) {
                return mid2;
            }
            if (iterable.get(mid1).compareTo(key)>0) {
                last_index = mid1 - 1;
            }
            else if (iterable.get(mid2).compareTo(key)<0) {
                first_index = mid2 + 1;
            }
            else {
                first_index = mid1 + 1;
                last_index = mid2 - 1;
            }
        }
        return -1;
    }
}

File: Sorts & Search & Shuffling Algorithms/Elementry Search/Ternary Search/IterativeTernary.py
class IterativeTernary:
    @staticmethod
    def Search(low, high, key, iterable):
        while high >= low:
            mid1 = low + (high - low) // 3
            mid2 = high - (high - low) // 3
            if key == iterable[mid1]:
                return mid1
            if key == mid2:
                return mid2
            if key < iterable[mid1]:
                high = mid1 - 1
            elif key > iterable[mid2]:
                low = mid2 + 1
            else:
                low = mid1 + 1
                high = mid2 - 1
        return -1